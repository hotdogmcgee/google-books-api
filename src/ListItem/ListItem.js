import React from 'react'


function ListItem(props) {

    const author = props.book.volumeInfo.authors ? 
        props.book.volumeInfo.authors[0]
        : "no author found";


    const price = props.book.saleInfo.retailPrice ?
        '$' + props.book.saleInfo.retailPrice["amount"]
        : "no price found"

    const thumbnail = props.book.volumeInfo.imageLinks ?
        props.book.volumeInfo.imageLinks.smallThumbnail
        : ""

    return (
        <div className="list-item">
            <h1>{props.book.volumeInfo.title}</h1>
            <div className="book-image"><img alt="book cover"src={thumbnail}></img></div>
            <div className="book-info">
                <div className="author-price-container">
                    <h2>author: {author}</h2>
                    <h2>price: {price}</h2>
                </div>
                <p>{props.book.volumeInfo.description}</p>
            </div>
        </div>
    )
}

export default ListItem