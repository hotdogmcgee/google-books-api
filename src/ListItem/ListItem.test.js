import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import ListItem from './ListItem';


describe('ListItem component', () => {
  const bookTest = {
    volumeInfo: {
      authors: ['bill'],
      imageLinks: {
        smallThumbnail: "image"
        },
      title: "Manhattan Beach",
      description: "lorem ipsum lalalallaa yep yep"
      },

    saleInfo: {
      retailPrice: {
        amount: 50
      }
    }
  }
    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ListItem book={bookTest}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected,', () => {
    const tree = renderer
      .create(<ListItem book={bookTest}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  })

})