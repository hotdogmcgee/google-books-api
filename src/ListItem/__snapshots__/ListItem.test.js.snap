// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component renders the UI as expected, 1`] = `
<div
  className="list-item"
>
  <h1>
    Manhattan Beach
  </h1>
  <div
    className="book-image"
  >
    <img
      alt="book cover"
      src="image"
    />
  </div>
  <div
    className="book-info"
  >
    <div
      className="author-price-container"
    >
      <h2>
        author: 
        bill
      </h2>
      <h2>
        price: 
        $50
      </h2>
    </div>
    <p>
      lorem ipsum lalalallaa yep yep
    </p>
  </div>
</div>
`;
